R/convert_sim_var.R

```diff
@@ -0,0 +1,70 @@
+
+#' convert an existing simulation variable into a different one
+#' 
+#' allows you to provide specific variable or unit conversions to create a new variable 
+#' that can be used with \code{\link{plot_var_compare}}, and others. 
+#' 
+#' @param nc_file the netcdf output file
+#' @param \dots an expression to convert variable based on other variables or offsets
+#' @param unit the units for the new variable
+#' @param longname the longname for the new variable
+#' @export
+#' @importFrom lazyeval lazy_dots lazy_eval
+#' @importFrom ncdf4 ncvar_def ncvar_put ncvar_add
+#' @examples 
+#' \dontrun{
+#'sim_folder <- run_example_sim(verbose = FALSE)
+#'nc_file <- file.path(sim_folder, 'output.nc')
+#'convert_sim_var(nc_file, tempF = temp/5*9+32, unit='degF',longname='temperature degrees Farenheit')
+#'plot_var(nc_file, 'tempF')
+#'convert_sim_var(nc_file, crazy_var = temp-u_mean*1000)
+#'plot_var(nc_file, 'crazy_var')
+#'
+#' }
+#' 
+convert_sim_var <- function(nc_file='output.nc', ..., unit='', longname=''){
```

Comment:
See comments on the PR. It doesn't now, but it could with a minor modification. Basically, would need to remove the `stop` where it checks the parsed variables for being in the netcdf and errors if they aren't. 

Instead it would only pull the vars in netcdf into the `data` object, and leave the other names alone. Then the function would be called normally.